backends/pipeline_api.py

```diff
@@ -30,7 +36,15 @@ def update_profile_from_edx(backend, user, response, is_new, *args, **kwargs):
 
     # this function is completely skipped if the backend is not edx or
     # the user has not created now
-    if backend.name != EdxOrgOAuth2.name or not is_new:
+    if backend.name != EdxOrgOAuth2.name:
+        return
+
+    if has_role(user, [Staff.ROLE_ID, Instructor.ROLE_ID]):
+        backend.strategy.session_set('next', '/learners')
+    else:
+        backend.strategy.session_set('next', '/dashboard')
+
+    if not is_new:
```

Comment:
why you are checking ``is_new`` at bottom? 
if user is not ``is_new`` then will he have roles?
what session_set for next do for not is_new users?
And what it do for new users?